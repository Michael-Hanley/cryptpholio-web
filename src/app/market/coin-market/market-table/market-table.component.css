.icon {
    width: 12px;
    height: 12px;
}
.mat-table {
    overflow: auto;
    min-width: 875px;
}
.mat-form-field {
    font-size: 14px;
    width: 100%;
}
.mat-column-icon {
    flex: 0 0 45px;
}
.mat-column-symbol {
    flex: 0 0 90px;
}
.mat-column-priceBtc {
    flex: 0 0 115px;
}
.mat-column-price {
    flex: 0 0 95px;
}

.mat-column-marketCap {
    flex: 0 0 155px;
}
.mat-column-priceChange1d {
    flex: 0 0 85px;
    justify-content: center;
}
.column {
    cursor: pointer;
}