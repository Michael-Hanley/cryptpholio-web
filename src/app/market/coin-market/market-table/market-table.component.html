<mat-table #table [dataSource]="dataSource" matSort>
    <ng-container matColumnDef="icon" >
        <mat-header-cell *matHeaderCellDef></mat-header-cell>
        <mat-cell class="column" *matCellDef="let coin" (click)='coinView(coin)'> <img src="{{coin.icon}}" width="30" height="30"> </mat-cell>
    </ng-container>
    <ng-container matColumnDef="name">
      <mat-header-cell *matHeaderCellDef> Name </mat-header-cell>
      <mat-cell class="column" *matCellDef="let coin" (click)='coinView(coin)'>{{coin.name}}</mat-cell>
    </ng-container>

    <ng-container matColumnDef="symbol">
      <mat-header-cell *matHeaderCellDef> Symbol </mat-header-cell>
      <mat-cell class="column" *matCellDef="let coin" (click)='coinView(coin)'>{{coin.symbol}}</mat-cell>
    </ng-container>

    <ng-container matColumnDef="price">
      <mat-header-cell *matHeaderCellDef mat-sort-header> Price USD </mat-header-cell>
      <mat-cell class="column" *matCellDef="let coin" (click)='coinView(coin)'>${{coin.price | number:'1.2-2'}}</mat-cell>
    </ng-container>

    <ng-container matColumnDef="priceBtc">
      <mat-header-cell *matHeaderCellDef mat-sort-header> Price BTC </mat-header-cell>
      <mat-cell class="column" *matCellDef="let coin" (click)='coinView(coin)'> <mat-icon class="icon" svgIcon="btc"></mat-icon>{{coin.priceBtc | number:'.2-9'}}</mat-cell>
    </ng-container>

    <ng-container matColumnDef="marketCap">
      <mat-header-cell *matHeaderCellDef mat-sort-header> Market Cap USD </mat-header-cell>
      <mat-cell class="column" *matCellDef="let coin" (click)='coinView(coin)'>${{coin.marketCap | number:'1.2-2'}}</mat-cell>
    </ng-container>
    
    <ng-container matColumnDef="priceChange1d">
      <mat-header-cell *matHeaderCellDef mat-sort-header> Percent Change 24h</mat-header-cell>
      <mat-cell class="column" *matCellDef="let coin" (click)='coinView(coin)'>{{coin.priceChange1d | number}}%</mat-cell>
    </ng-container>
    

    <ng-container matColumnDef="availableSupply">
      <mat-header-cell *matHeaderCellDef mat-sort-header> Available Supply </mat-header-cell>
      <mat-cell class="column" *matCellDef="let coin" (click)='coinView(coin)'>{{coin.availableSupply}}</mat-cell>
    </ng-container>

    <mat-header-row *matHeaderRowDef="columnsToDisplay"></mat-header-row>
    <mat-row *matRowDef="let coins; columns: columnsToDisplay"></mat-row>
  </mat-table>