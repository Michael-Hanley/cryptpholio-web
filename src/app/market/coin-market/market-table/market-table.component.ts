import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
import { Router } from '@angular/router';

@Component({
  selector: 'app-market-table',
  templateUrl: './market-table.component.html',
  styleUrls: ['./market-table.component.scss']
})
export class MarketTableComponent implements OnInit {
  @Input() dataSource;
  @Input() coins;
  @Input() columnsToDisplay;
  @Output() navigated = new EventEmitter<boolean>();

  constructor(private router: Router) { }

  coinView(coin) {
    this.navigated.emit(true);
    this.router.navigate(['/coin'], {queryParams: {coinId: coin.id}});
  }

  ngOnInit() {
  }

}
