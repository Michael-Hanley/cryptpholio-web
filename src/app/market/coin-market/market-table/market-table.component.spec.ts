import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';

import { MarketTableComponent } from './market-table.component';

import { MatTableModule } from '@angular/material/table';
import { MatIconModule } from '@angular/material/icon';
import { Router } from '@angular/router';


describe('MarketTableComponent', () => {
  let component: MarketTableComponent;
  let fixture: ComponentFixture<MarketTableComponent>;
  let router: Router;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ MarketTableComponent ],
      imports: [
        MatTableModule,
        MatIconModule,
        RouterTestingModule.withRoutes([])
      ]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(MarketTableComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();

    router = TestBed.get(Router);
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
  it('should navigate', () => {
    const navigateSpy = spyOn(router, 'navigate');
    const coin = {
      id: 'test-coin'
    };
    component.coinView(coin);

    expect(navigateSpy).toHaveBeenCalledWith(['/coin'], { queryParams: { coinId: 'test-coin' } });
  });
  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
