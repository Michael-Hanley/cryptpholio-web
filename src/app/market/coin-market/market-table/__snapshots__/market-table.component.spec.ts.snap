// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarketTableComponent should match snapshot 1`] = `
<app-market-table
  navigated={[Function EventEmitter]}
  router={[Function Router]}
>
  <mat-table
    class="mat-table"
    matsort=""
    role="grid"
  >
    
    <mat-header-row
      class="mat-header-row"
      role="row"
    >
      
    </mat-header-row>
    
    
  </mat-table>
</app-market-table>
`;
